src/SFA.DAS.ProviderApprenticeshipsService.Web/Orchestrators/CommitmentOrchestrator.cs

```diff
@@ -404,88 +351,58 @@ public class CommitmentOrchestrator : BaseCommitmentOrchestrator
             {
                 ProviderId = providerId,
                 HashedCommitmentId = hashedCommitmentId,
-                LegalEntityName = data.Commitment.LegalEntityName,
-                Reference = data.Commitment.Reference,
-                Status = data.Commitment.GetStatus(),
+                LegalEntityName = commitment.LegalEntityName,
+                Reference = commitment.Reference,
+                Status = commitment.GetStatus(),
                 HasApprenticeships = apprenticeships.Count > 0,
                 Apprenticeships = apprenticeships,
-                LatestMessage = GetLatestMessage(data.Commitment.Messages, true)?.Message,
-                PendingChanges = data.Commitment.AgreementStatus != AgreementStatus.EmployerAgreed,
+                LatestMessage = GetLatestMessage(commitment.Messages, true)?.Message,
+                PendingChanges = commitment.AgreementStatus != AgreementStatus.EmployerAgreed,
                 ApprenticeshipGroups = apprenticeshipGroups,
                 RelationshipVerified = relationshipRequest.Relationship.Verified.HasValue,
-                IsReadOnly = data.Commitment.EditStatus != EditStatus.ProviderOnly,
-                IsFundedByTransfer = data.Commitment.TransferSender != null,
+                IsReadOnly = commitment.EditStatus != EditStatus.ProviderOnly,
+                IsFundedByTransfer = commitment.IsTransfer(),
                 Errors = errors,
                 Warnings = warnings
             };
         }
 
         public async Task<DeleteCommitmentViewModel> GetDeleteCommitmentModel(long providerId, string hashedCommitmentId)
         {
-            var commitmentId = _hashingService.DecodeValue(hashedCommitmentId);
+            var commitment = await GetCommitment(providerId, hashedCommitmentId);
 
-            var data = await _mediator.SendAsync(new GetCommitmentQueryRequest
-            {
-                ProviderId = providerId,
-                CommitmentId = commitmentId
-            });
-
-            Func<string, string> textOrDefault = txt => !string.IsNullOrEmpty(txt) ? txt : "without training course details";
+            string TextOrDefault(string txt) => !string.IsNullOrEmpty(txt) ? txt : "without training course details";
 
-            var programmeSummary = data.Commitment.Apprenticeships
+            var programmeSummary = commitment.Apprenticeships
```

Comment:
Could the Apprenticeships be null?

Reply:
Same as previous comments - we could do better, but it's not new or changed behaviour, so can be left for future refactoring.